pkg/controllers/termination/terminate.go

```diff
@@ -153,8 +149,8 @@ func (t *Terminator) evict(ctx context.Context, pods []*v1.Pod) {
 }
 
 func IsStuckTerminating(pod *v1.Pod) bool {
-	if pod.DeletionTimestamp.IsZero() {
+	if pod.DeletionTimestamp == nil {
 		return false
 	}
-	return time.Since(pod.DeletionTimestamp.Time) > time.Duration(ptr.Int64Value(pod.DeletionGracePeriodSeconds))*time.Second
+	return injectabletime.Now().After(pod.DeletionTimestamp.Time)
```

Comment:
I think we want to check the `DeletionGracePeriodSeconds` here. Evictions don't always take some `ms` amount of time, and we want to make sure we don't consider it a pod that `IsStuckTerminating` if it's just taking longer to evict.

Additionally, this would essentially always be true whenever the `DeletionTimestamp` is non-nil, boiling it down to a `return true`. `DeletionTimestamp` is always nil if it hasn't been told to delete, and we would only check if it `IsStuckTerminating()` after we have told it to delete. Thus, since this logically comes after deleting a pod, this will always be true.

Reply:
Discussed offline.